### ISSUE TITLE
Recommended distro/version??

### ISSUE THREAD

**Issue by pepa65** [bill-mavromatis/gnome-layout-manager#15]
Is this especially meant to work on Ubuntu? (Which version is especially well tested??) Or is it for any distro? Any versions of Gnome3 where this doesn't work well, or works especially well?

**Comment by vmavromatis** [comment]
Hi @pepa65! So far I have tested it on the following:
- Manjaro with GNOME 3.22/3.24
- Antergos with GNOME  3.24 		
- Fedora with GNOME 3.22 	
- Ubuntu with GNOME 3.24 		
- Arch Linux with GNOME 3.24 

The most tested version is Manjaro and it's the one I know 100% works (it's my system so I test it often). Last I checked Ubuntu should be fine, but I will check again with its latest version.

**Comment by aviwad** [comment]
I use it regularly on Ubuntu gnome 17.04, no issues ;) :wink:

**Comment by pepa65** [comment]
Thanks for your reply!
I used Ubuntu 17.04 (Gnome 3.24) and it works well. (Fairly fast too..!)

As a switcher between these layouts it doesn't work as well, because it tries to (re-)download resources that should already have been installed.

**Comment by aviwad** [comment]
Well, thats true. Thats why if i put my laptop on "Aeroplane Mode" it never works :frowning_face:

**Comment by vmavromatis** [comment]
You are correct guys, what happens now is that it goes to extensions.gnome.org first to get the extension names and then checks if they are installed. I will fix this! Flagging it as enhancement.

**Comment by aviwad** [comment]
WOW! Talk about fast replies! one hell of a damn cool dev!

**Comment by vmavromatis** [comment]
@GNOME-IS-LIFE @pepa65 Latest version should work offline if you have everything installed. I changed many things and made the code much faster. Please let me know what you think.

PS: I also added Global Menu for Unity, so if you want to use that, make sure you install unity gtk module! I have instructions on the main page.

**Comment by pepa65** [comment]
The Global Menu will be a welcome addition! Can't you just integrate the install command to the script?

**Comment by vmavromatis** [comment]
I could, but then it would require admin privileges to run the script (root) and many people are very skeptical about running scripts that ask for root. However, I'm considering adding them since there are some very cool modules that can fit the project so well (Plotinus and nocsd).

Edit: I just had an idea. What if I had an extra option "Install all Requirements (requires root)". Or better to integrate it inside the Unity layout option?

**Comment by pepa65** [comment]
If you put the sudo in the script, it will only ask for the password when it need it / gets to it, and in that context people will understand what the password is for. ;-)

**Comment by vmavromatis** [comment]
You're not wrong, I'll see what I can do.

**Comment by incrazyboyy** [comment]
You could use gksudo instead of sudo, it shows a graphical password promt including the command which will be run as root!

**Comment by Mikhus** [comment]
After upgrade to Ubuntu 19.10 from 18.04, sadly, but it does not work any more. After applying any layout it produces splash screen telling "Something went wrong" and asking for re-login. After re-login themes only partially applied. But unity-like desktop layout sticks forever... On 18.04 everything was working just perfectly...

Kernel: 5.5.7-050507-generic
GNOME: 3.34.2
Ubuntu: 19.10 eoan
